src/UKHO.ADDS.EFS.Builder.S100/Pipelines/Assemble/ProductSearchNode.cs

```diff
@@ -0,0 +1,208 @@
+                if (!string.IsNullOrEmpty(queryString))
+                {
+                    var parsedValues = ParseQueryString(queryString);
+                    limit = parsedValues.TryGetValue("limit", out var urlLimit) ? int.Parse(urlLimit) : limit;
+                    start = parsedValues.TryGetValue("start", out var urlStart) ? int.Parse(urlStart) : start;
+                    filter = parsedValues.TryGetValue("$filter", out var urlFilter) ? urlFilter : filter;
+                }
+                else
+                {
+                    filter = string.Empty;
+                }
+
+            } while (batchSearchResponse.Entries.Count < totalUpdateCount && !string.IsNullOrWhiteSpace(filter));
+
+            batchSearchResponse.Count = queryCount;
+            return batchSearchResponse;
+        }
+
+        private static IEnumerable<List<T>> SplitList<T>(List<T> masterList, int size = 30)
+        {
+            for (var i = 0; i < masterList.Count; i += size)
+            {
+                yield return masterList.GetRange(i, Math.Min(size, masterList.Count - i));
+            }
+        }
+
+        private string GenerateQueryForFss(List<SearchBatchProducts> products)
+        {
+            var queryBuilder = new StringBuilder();
+            var logBuilder = new StringBuilder();
+
+            if (products == null || products.Count == 0)
+                return string.Empty;
+
+            queryBuilder.Append('(');
+            for (var i = 0; i < products.Count; i++)
+            {                
+                var product = products[i];
+                queryBuilder.Append('(')
+                    .AppendFormat(_fileShareServiceConfiguration.ProductName ?? string.Empty, product.ProductName)
+                    .AppendFormat(_fileShareServiceConfiguration.EditionNumber ?? string.Empty, product.EditionNumber);
+
+                if (product.UpdateNumbers != null && product.UpdateNumbers.Any())
+                {
+                    queryBuilder.Append("((");
+                    queryBuilder.Append(string.Join(" or ", product.UpdateNumbers.Select(u => string.Format(_fileShareServiceConfiguration.UpdateNumber ?? string.Empty, u))));
+                    queryBuilder.Append("))");
+                }
+                queryBuilder.Append(i == products.Count - 1 ? ")" : ") or ");
+                logBuilder.AppendFormat("\n Product/ProductName:{0}, EditionNumber:{1}, UpdateNumbers:[{2}]",
+                    product.ProductName,
+                    product.EditionNumber,
+                    string.Join(",", product?.UpdateNumbers?.Where(u => u.HasValue) ?? []));
+            }
+            queryBuilder.Append(')');
+            _logger.LogProductSearchNodeFssQueryProducts(logBuilder.ToString());
+            return (queryBuilder.ToString());
+        }
+
+        private List<SearchBatchProducts> SliceProductsWithUpdateNumberForFssQuery(List<SearchBatchProducts> products)
```

Comment:
naming suggestion for clarity and Use IEnumerable<T> for method parameters and return types when you only enumerate the collection. 
```suggestion
        private List<SearchBatchProducts> ChunkProductsByUpdateNumberLimit(IEnumerable<SearchBatchProducts> products)
```

Reply:
method rename done